src/js/actions/superselect.js

```diff
@@ -353,20 +371,20 @@ define(function (require, exports) {
             .then(function (hitLayerIDs) {
                 var clickedSelectableLayerIDs,
                     coveredLayers = _getContainingLayerBounds.call(this, layerTree, coords.x, coords.y),
-                    coveredLayerIDs = collection.pluck(coveredLayers, "id").concat(hitLayerIDs);
+                    hitLayerIDsSansArtboards = _removeArtboardIDs(layerTree, Immutable.List(hitLayerIDs)),
+                    coveredLayerIDs = collection.pluck(coveredLayers, "id").concat(hitLayerIDsSansArtboards);
 
                 coveredLayerIDs = coveredLayerIDs.sortBy(function (id) {
                     return hitLayerIDs.indexOf(id);
                 });
 
                 if (deep) {
-                    // Select any non-group layer
+                    // Select any non-group layer, and allow for artboard badges
                     var hitLayerMap = coveredLayerIDs.reduce(function (layerMap, id) {
```

Comment:
Seem like this doesn't really need to be a faux-map anymore.  It probably don't hurt anything, but might be more clear as a simple array of Ids, since all it is used for (I think) is testing for existence (?)